### PR TITLE
Modernize weather template

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Modernize weather template

Use the new config options of forecast instead of the old legacy forecast state attribute.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/98064
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/weather.template.markdown [hunk 1]
```diff
@@ -43,7 +43,7 @@ weather:
 name:
   description: Name to use in the frontend.
   required: true
-  type: string
+  type: template
```
Comment (jbouwh) [inline source/_integrations/weather.template.markdown:46]: I dos not seem the `name` property accepts a template value refering to your PR. Or am I wrong?

**Review by jbouwh — CHANGES_REQUESTED** [review]

File: source/_integrations/weather.template.markdown (hunk 1 above)
Comment (gjohansson-ST) [reply source/_integrations/weather.template.markdown:46]: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/template_entity.py#L66

Name is actually a template but in most places it's mentioned as a string.
For this PR I took the liberty of changing it for this platform.

Comment (jbouwh) [reply source/_integrations/weather.template.markdown:46]: I did not see cv.ensure_temlate in the schema. So it seem that is added later.

**Review by jbouwh — APPROVED** [review]

**Comment by jbouwh** [comment]
Can be merged when the parent PR gets merged.

**Pull request merged**